src/vaadin-custom-field-mixin.html

```diff
@@ -150,19 +150,25 @@
       this.__settingValue = false;
     }
 
+    /* Like querySelectorAll('*') but also gets all elements through any nested slots recursively */
+    __queryAllAssignedElements(elem) {
+      const result = [];
+      let elements;
+      if (elem.tagName === 'SLOT') {
+        elements = elem.assignedNodes({flatten: true}).filter(node => node.nodeType === Node.ELEMENT_NODE);
+      } else {
+        result.push(elem);
+        elements = Array.from(elem.children);
+      }
+      elements.forEach(elem => {
+        result.push(...this.__queryAllAssignedElements(elem));
+      });
+      return result;
+    }
```

Comment:
Here's an alternate version of the same method that does exactly the same thing (and also passes all tests). I'm just not sure which version I prefer. I think the committed one (above) is more readable (maybe) even though it uses temporary variables (`result` and `elements`).

```js
__queryAllAssignedElements(elem) {
  const flatten = (a, b) => a.concat(b);
  if (elem.tagName === 'SLOT') {
    return elem.assignedNodes({flatten: true})
      .filter(node => node.nodeType === Node.ELEMENT_NODE)
      .map(this.__queryAllAssignedElements.bind(this))
      .reduce(flatten);
  } else {
    return Array.from(elem.children)
      .map(this.__queryAllAssignedElements.bind(this))
      .reduce(flatten, [elem]);
  }
}
```

Reply:
I would prefer the first option (that is already in the code) as it's more readable.